src/corehost/cli/fxr/sdk_resolver.cpp

```diff
@@ -179,16 +179,22 @@ bool sdk_resolver_t::resolve_sdk_dotnet_path(const pal::string_t& dotnet_root, c
     std::vector<pal::string_t> global_dirs;
     bool multilevel_lookup = multilevel_lookup_enabled();
 
-    if (!dotnet_root.empty())
-    {
-        hive_dir.push_back(dotnet_root);
-    }
+    // dotnet_root contains DIR_SEPARATOR appended that we need to remove.
+    assert(!dotnet_root.empty());
```

Comment:
IIRC, I think it can be empty if passed in that way from hostfxr_resolve_sdk. I think I even documented that you can do this to use only the multilevel locations. I have to double check the code, though.

Reply:
Yes I see the doc for hostfxr_resolve_sdk does say that. I'll add the check back and adjust the check that removes the last slash